include/nbl/asset/utils/CSPIRVIntrospector.h

```diff
@@ -648,35 +648,30 @@ class NBL_API2 CSPIRVIntrospector : public core::Uncopyable
 			if (layout)
 			{
 				// TODO: merge CStageIntrospection into CPipelineIntrospection
+				pplnIntrospectData->merge(introspection.get());
```

Comment:
you can probably hoist the `merge` above the if statement.

You need to handle the failure case of an unsuccessful `merge` and return nullptr.

Reply:
then for the given `layout` you need to compare for compatibility, basically "are all the bindings in `pplnIntrospectData` present `layout` with same type and array size? (Also, if present, is the runtime sized/variable length array the last binding of both `pplnIntrospectData` and `layout`)"